extensions/amp-next-page/amp-next-page.md

```diff
@@ -25,237 +25,272 @@ limitations under the License.
-next page fails to load), a box will be displayed with links to the next three
-pages. This box will also be displayed after the maximum number of articles
-have been rendered.
+The `<amp-next-page>` component renders the footer recommendation box if one of the following situations arise:
 
-The appearance of these links can be customized by styling the following
-classes:
+- The user reaches the end of the page before the next page had loaded.
+- The next page fails to load.
+- If the `max-pages` attribute is specified and the number of displayed pages is met.
 
-- `.amp-next-page-links` for the containing element
-- `.amp-next-page-link` for an individual link
-- `.amp-next-page-image` for the link image
-- `.amp-next-page-text` for the link text
+The footer recommendation box contains links to the remaining pages. The default footer recommendation box renders the specified image and title used in the JSON configuration and can be styled as specified in the Styling section.
+
+The footer recommendation box can also be provided as a custom component inside `<amp-next-page>` as any element that has the `footer`. It can also be templated via `amp-mustache` or other templating engines, in which case it will be passed an array of the remaining `pages`, each being an object with a `title`, `url` and `image`. Example:
+
+```html
+<amp-next-page src="https://example.com/config.json">
+  <div footer class="my-custom-footer">
+    Here are a few more articles:
+    <template type="amp-mustache">
+      <div class="footer-content">
+        {{#pages}}
+        <span class="title">{{title}}</span>
+        <span class="url">{{url}}</span>
+        <span class="image">{{image}}</span>
+        {{/pages}}
+      </div>
+    </template>
+  </div>
+</amp-next-page>
+```
 
 ### Separator
 
 A separator is rendered between each loaded document. By default this is
-rendered as a full-width hairline. It can be customised by styling the `.amp-next-page-default-separator` class.
+rendered as a full-width hairline. Refer to the Styling section for information on customizing this default behavior.
 
 Alternatively, you can specify a custom separator containing arbitrary HTML
 content as a child of the `amp-next-page` component by using the `separator`
-attribute.
+attribute. Similar to the footer box, custom separator can be templated via `amp-mustache` or other templating engines, and will be passed the `title`, `url` and `image` of the upcoming article.
 
 ```html
 <amp-next-page src="https://example.com/config.json">
-  <div separator>
-    <h1>Keep reading</h1>
+  <div separator class="my-custom-separator">
+    <template type="amp-mustache">
+      <div class="my-custom-separator-content">
+        <amp-img
+          src="{{image}}"
+          layout="fixed"
+          height="16"
+          width="16"
+        ></amp-img>
+        <span>Next article: {{title}}</span>
+      </div>
+    </template>
   </div>
 </amp-next-page>
 ```
```

Comment:
Most of this section should be updated and moved under styling. I've added a comment that updates this.